prboom2/src/dsda/utility/string_view.h

```diff
@@ -0,0 +1,58 @@
+//
+// Copyright(C) 2023 by Ryan Krafnick
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// DESCRIPTION:
+//	DSDA Utility String View
+//
+
+#ifndef __DSDA_UTILITY_STRING_VIEW__
+#define __DSDA_UTILITY_STRING_VIEW__
+
+#include "doomtype.h"
+
+// Non-allocated, read-only view into a string, like C++ string_view
+typedef struct {
+  const char* string;
+  size_t size;
+} dsda_string_view_t;
+
+void dsda_InitStringView(dsda_string_view_t* sv, const char* string, size_t size);
+
+dboolean dsda_IsStringViewEmpty(const dsda_string_view_t* sv);
+
+void dsda_StringViewAtOffset(const dsda_string_view_t* sv, size_t offset, dsda_string_view_t* ofs);
+
+// Splits `sv` at the first instance of `c`, putting everything up to and including it in `before`
+// and everything after in `after`.  `before` or `after` may safely alias `sv`.  If `c` does not
```

Comment:
I'm confused by the use of alias in this file. What does it mean in this context?

Reply:
That's C jargon for "point to the same object in memory".

Reply:
Ah now I see what you mean. In other words you can safely pass the source as the before / after as well.